blockchair/src/endpoint/balance.ts

```diff
@@ -0,0 +1,147 @@
+import objectPath from 'object-path'
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { AdapterRequest } from '@chainlink/types'
+import { Config, DEFAULT_DATA_PATH, getBaseURL } from '../config'
+import { CoinType, ChainType } from '.'
+
+export const Name = 'balance'
+
+type Address = {
+  address: string
+  coin?: CoinType
+  chain?: ChainType
+  balance?: number
+  warning?: string
+}
+
+type AddressGroup = {
+  addresses: string[]
+  result: Address[]
+}
+
+type GroupedAddresses = {
+  [network: string]: AddressGroup
+}
+
+type ResponseWithResult = {
+  response: any
+  result: Address[]
+}
+
+type RequestData = {
+  dataPath: string
+  confirmations: number
+}
+
+const coins: { [ticker: string]: string } = {
+  btc: 'bitcoin',
+  dash: 'dash',
+  doge: 'dogecoin',
+  ltc: 'litecoin',
+  zec: 'zcash',
+  bch: 'bitcoin-cash',
+  bsv: 'bitcoin-sv',
+  grs: 'groestlcoin',
+}
+
+const WARNING_NO_OPERATION_TESTNET =
+  'No Operation: only testnet supported by blockcypher adapter is btc'
+const WARNING_NO_OPERATION_MISSING_ADDRESS = 'No Operation: address param is missing'
+const WARNING_NO_OPERATION_ADDRESS_NOT_FOUND = 'No Operation: address does not exist on network'
+
+const group = (addresses: Address[]) => {
+  const output: GroupedAddresses = {}
+  for (const addr of addresses) {
+    if (!addr?.address) addr.warning = WARNING_NO_OPERATION_MISSING_ADDRESS
+    if (!addr?.coin) {
+      addr.coin = 'btc'
+    }
+    if (!addr?.chain) {
+      addr.chain = 'mainnet'
+    }
+    const { address, coin, chain } = addr
+    if (output[`${coin}-${chain}`]) {
+      output[`${coin}-${chain}`].addresses.push(address)
+      output[`${coin}-${chain}`].result.push(addr)
+    } else output[`${coin}-${chain}`] = { addresses: [address], result: [addr] }
```

Comment:
This can be made more readable like this:
```suggestion
    const key = `${coin}-${chain}`
    output[key] = output[key] ||  { addresses: [], result: [] }
    output[key].addresses.push(address)
    output[key].result.push(addr)
```